File: go/internal/services/post/post.go
package post

import (
	"time"
)

type Post struct {
	ID       string    `json:"id" bson:"_id,omitempty"`
	Title    string    `json:"title" bson:"title,omitempty"`
	Body     string    `json:"body" bson:"body,omitempty"`
	DateTime time.Time `json:"date_time" bson:"date_time,omitempty"`
}

File: java/spring-boot/src/main/java/com/example/demo_mongo/post/Post.java
package com.example.demo_mongo.post;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.UUID;

@Document("post")
public record Post(
        @Id
        String id,
        String title,
        String body,
        Date dateTime
) {
	public Post(String title) {
		this(UUID.randomUUID().toString(), title, null, null);
	}
}
